legal-api/src/legal_api/resources/v2/business/business.py

```diff
@@ -164,7 +164,13 @@ def search_businesses():
         draft_query = db.session.query(Filing).filter(
             and_(Filing.temp_reg.in_(identifiers), Filing.business_id.is_(None)))
 
-        bus_results = [x.json(slim=True) for x in bus_query.all()]
+        bus_results = []
+        for business in bus_query.all():
+            business_json = business.json(slim=True)
+            # add alternateNames array to slim json only to firms
+            if business.legal_type in (Business.LegalTypes.SOLE_PROP, Business.LegalTypes.PARTNERSHIP):
```

Comment:
So, after investigating and looking at the code in AUTH WEB, I believe we don't need to have the alternateNames array for everything. We only show the alternate name in MBR for firms. Otherwise, we show legal name. This also helped boost performance by a lot.

Reply:
You'll need to verify with @kialj876 whether he uses `alternateNames` at all for non-firms for the search endpoint.

If we remove this, it will be hard to maintain as it is a weird edge case and I believe the search api users are expecting alternate names for non-firms as well where it's applicable.

Reply:
Very good point, makes a lot of sense.

My thinking was, when we first implemented the easy legal name fix, the alternateNames were in big JSON. It was only when we needed to fix the names in MBR that we decided to move it to slim so that it's fixed here in the search endpoint.

Reply:
I'm not using this endpoint for anything, but I think auth uses it in the proxy call for the 'My Business Registry' table so check with @seeker25

Reply:
Only usage here:
![image](https://github.com/bcgov/lear/assets/3484109/45e08ebc-3e2c-4a87-836c-a253f8c7b465)

Reply:
Only SP and GP

Reply:
ok, for some reason I thought we might be using it outside of MBR.

think we can go with this change.  

Can you add a note the search endpoint to make note that we are not returning alternate_names for non-firms due to perf issues?

Reply:
FYI this is what I am talking about: https://github.com/bcgov/sbc-auth/blob/main/auth-api/src/auth_api/resources/v1/org.py#L355 

^ is the call effected by this change and you won't be able to determine the 'operating name' anymore in that table. 
Pretty sure this change would cause the table to display 'Hrvoje..' instead


![Screenshot 2024-07-04 at 11 51 22 AM](https://github.com/bcgov/lear/assets/16654897/bfaab876-a423-48b8-a653-2131fbe63e70)

Reply:
Since this is an SP, the alternateNames array will be there (including the name). So, Auth Web will be able to get it and output it correctly I believe. I can make the call locally and show.

Reply:
oops nvm only sp / gp as travis said -- sorry github didn't update with all your comments for me till after haha

Reply:
All good, no worries 🙏